### DISCUSSION TITLE
Regex to search Socket ID

### DISCUSSION THREAD

**Discussion by saurabh-lodha-16** [socketio/socket.io#5170] [Q&A] [answered]
I have integrated Socket.io and use it for client and server data communication.
```
 socket.join(room); // join the room
 await redis.set(socket.id, room); // update redis with socket.id->room values
 ```
 
 The problem is I did not add an expiry here and now my Redis is flooded with socket IDs. Since my database has other info, I wnated to know if there is an regex pattern for socket.id which I can use to search and then delete.

**Comment by darrachequesne** [comment, marked as answer]
Hi!

You can use the following regex: `[a-zA-Z0-9_-]{20}` (or `[\w-]{20}`)

**Comment by saurabh-lodha-16** [comment]
Thanks a lot will try it.
